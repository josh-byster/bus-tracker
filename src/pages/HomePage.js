import React, { Component } from "react";
import "../styles/home.scss";
import StopSearch from "../components/StopSearch";
class HomePage extends Component {
  render() {
    return (
      <div>
        <div className="home d-flex">
          <div className="info-box">
            <h1>UIUC Bus Tracker</h1>
            <StopSearch />
          </div>
        </div>
        <footer className="text-muted">
          <div className="container">
            <p>
              Copyright © 2019 UIUCBus.com
              <br />
              Data provided by{" "}
              <a href="https://mtd.org/">
                Champaign—Urbana Mass Transit District
              </a>
              .
              <br />
              Suggestions, comments, or want to help contribute?{" "}
              <a href="https://github.com/josh-byster/bus-tracker">
                Check it out on GitHub!
              </a>
            </p>
          </div>
        </footer>
      </div>
    );
  }
}

export default HomePage;
