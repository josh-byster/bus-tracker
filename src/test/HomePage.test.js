import React from "react";
import expect from "expect";
import { shallow, mount } from "enzyme";
import EnzymeToJson from "enzyme-to-json";
// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import HomePage from "../pages/HomePage";

configure({ adapter: new Adapter() });

describe("Bus Info Modal Component", () => {
  const elem = mount(<HomePage />);

  it("renders the proper page", () => {
    expect(EnzymeToJson(elem)).toMatchSnapshot();
  });
});
