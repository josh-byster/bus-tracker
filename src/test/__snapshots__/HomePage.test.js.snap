// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bus Info Modal Component renders the proper page 1`] = `
<HomePage>
  <div>
    <div
      className="home d-flex"
    >
      <div
        className="info-box"
      >
        <h1>
          UIUC Bus Tracker
        </h1>
        <StopSearch>
          <div
            className="search-wrapper"
          >
            <Autosuggest
              alwaysRenderSuggestions={false}
              className="form-control"
              focusInputOnSuggestionClick={true}
              getSuggestionValue={[Function]}
              highlightFirstSuggestion={true}
              id="1"
              inputProps={
                Object {
                  "onChange": [Function],
                  "placeholder": "Type the name of a stop",
                  "value": "",
                }
              }
              multiSection={false}
              onSuggestionSelected={[Function]}
              onSuggestionsClearRequested={[Function]}
              onSuggestionsFetchRequested={[Function]}
              renderSuggestion={[Function]}
              renderSuggestionsContainer={[Function]}
              shouldRenderSuggestions={[Function]}
              suggestions={Array []}
              theme={
                Object {
                  "container": "react-autosuggest__container",
                  "containerOpen": "react-autosuggest__container--open",
                  "input": "react-autosuggest__input",
                  "inputFocused": "react-autosuggest__input--focused",
                  "inputOpen": "react-autosuggest__input--open",
                  "sectionContainer": "react-autosuggest__section-container",
                  "sectionContainerFirst": "react-autosuggest__section-container--first",
                  "sectionTitle": "react-autosuggest__section-title",
                  "suggestion": "react-autosuggest__suggestion",
                  "suggestionFirst": "react-autosuggest__suggestion--first",
                  "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
                  "suggestionsContainer": "react-autosuggest__suggestions-container",
                  "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
                  "suggestionsList": "react-autosuggest__suggestions-list",
                }
              }
            >
              <Autowhatever
                containerProps={Object {}}
                getSectionItems={[Function]}
                highlightedItemIndex={null}
                highlightedSectionIndex={null}
                id="1"
                inputProps={
                  Object {
                    "onBlur": [Function],
                    "onChange": [Function],
                    "onFocus": [Function],
                    "onKeyDown": [Function],
                    "placeholder": "Type the name of a stop",
                    "value": "",
                  }
                }
                itemProps={[Function]}
                items={Array []}
                multiSection={false}
                renderInputComponent={[Function]}
                renderItem={[Function]}
                renderItemData={
                  Object {
                    "query": "",
                  }
                }
                renderItemsContainer={[Function]}
                renderSectionTitle={[Function]}
                theme={
                  Object {
                    "container": "react-autosuggest__container",
                    "containerOpen": "react-autosuggest__container--open",
                    "input": "react-autosuggest__input",
                    "inputFocused": "react-autosuggest__input--focused",
                    "inputOpen": "react-autosuggest__input--open",
                    "item": "react-autosuggest__suggestion",
                    "itemFirst": "react-autosuggest__suggestion--first",
                    "itemHighlighted": "react-autosuggest__suggestion--highlighted",
                    "itemsContainer": "react-autosuggest__suggestions-container",
                    "itemsContainerOpen": "react-autosuggest__suggestions-container--open",
                    "itemsList": "react-autosuggest__suggestions-list",
                    "sectionContainer": "react-autosuggest__section-container",
                    "sectionContainerFirst": "react-autosuggest__section-container--first",
                    "sectionTitle": "react-autosuggest__section-title",
                  }
                }
              >
                <div
                  aria-expanded={false}
                  aria-haspopup="listbox"
                  aria-owns="react-autowhatever-1"
                  className="react-autosuggest__container"
                  key="react-autowhatever-1-container"
                  role="combobox"
                >
                  <input
                    aria-activedescendant={null}
                    aria-autocomplete="list"
                    aria-controls="react-autowhatever-1"
                    autoComplete="off"
                    className="react-autosuggest__input"
                    key="react-autowhatever-1-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Type the name of a stop"
                    type="text"
                    value=""
                  />
                  <div
                    className="react-autosuggest__suggestions-container"
                    id="react-autowhatever-1"
                    key="react-autowhatever-1-items-container"
                    role="listbox"
                  />
                </div>
              </Autowhatever>
            </Autosuggest>
            <Button
              className="location-btn"
              color="secondary"
              onClick={[Function]}
              tag="button"
            >
              <button
                aria-label={null}
                className="location-btn btn btn-secondary"
                onClick={[Function]}
                type="button"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        384,
                        512,
                        Array [],
                        "f3c5",
                        "M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z",
                      ],
                      "iconName": "map-marker-alt",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-map-marker-alt fa-w-12 "
                    data-icon="map-marker-alt"
                    data-prefix="fas"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 384 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
              </button>
            </Button>
            <Geolocated(NearestStop)
              isOpen={false}
              toggle={[Function]}
            >
              <NearestStop
                coords={null}
                isGeolocationAvailable={false}
                isGeolocationEnabled={true}
                isOpen={false}
                positionError={null}
                toggle={[Function]}
              >
                <div>
                  <Modal
                    autoFocus={true}
                    backdrop={true}
                    backdropTransition={
                      Object {
                        "mountOnEnter": true,
                        "timeout": 150,
                      }
                    }
                    centered={false}
                    className="nearest-stop-modal"
                    fade={true}
                    isOpen={false}
                    keyboard={true}
                    modalTransition={
                      Object {
                        "timeout": 300,
                      }
                    }
                    onClosed={[Function]}
                    onOpened={[Function]}
                    role="dialog"
                    toggle={[Function]}
                    zIndex={1050}
                  />
                </div>
              </NearestStop>
            </Geolocated(NearestStop)>
          </div>
        </StopSearch>
      </div>
    </div>
    <footer
      className="text-muted"
    >
      <div
        className="container"
      >
        <p>
          Copyright Â© 2019 UIUCBus.com
          <br />
          Data provided by
           
          <a
            href="https://mtd.org/"
          >
            ChampaignâUrbana Mass Transit District
          </a>
          .
          <br />
          Suggestions, comments, or want to help contribute?
           
          <a
            href="https://github.com/josh-byster/bus-tracker"
          >
            Check it out on GitHub!
          </a>
        </p>
      </div>
    </footer>
  </div>
</HomePage>
`;
